code/game/objects/effects/anomalies/anomalies_ectoplasm.dm

```diff
@@ -0,0 +1,211 @@
+/obj/effect/anomaly/ectoplasm
+	name = "ectoplasm anomaly"
+	desc = "It looks like the souls of the damned are trying to break into the realm of the living again. How upsetting."
+	icon_state = "ectoplasm"
+	aSignal = /obj/item/assembly/signaler/anomaly/ectoplasm
+	lifespan = 100 SECONDS //This one takes slightly longer, because it can run away.
+
+	///Blocks the anomaly from updating ghost count. Used in case an admin wants to rig the anomaly to be a certain size or intensity.
+	var/override_ghosts = FALSE
+	///The numerical power of the anomaly. Calculated in anomalyEffect. Also used in determining the category of detonation effects.
+	var/effect_power = 0
+	///The actual number of ghosts orbiting the anomaly.
+	var/ghosts_orbiting = 0
+
+/obj/effect/anomaly/ectoplasm/Initialize(mapload, new_lifespan, drops_core)
+	. = ..()
+
+	AddComponent(/datum/component/deadchat_control/cardinal_movement, ANARCHY_MODE, list(), 7 SECONDS)
+
+	if(. == COMPONENT_INCOMPATIBLE)
+		return
```

Comment:
Why is this check at the bottom of the Initialize proc, what's it's function? Having a return at the dead end of Initialize doesn't do anything.